include/utils.h

```diff
@@ -31,6 +31,9 @@ typedef int FileHandle;
 #include "memory_mapped_files.h"
 #endif
 
+#undef max
```

Comment:
Why is this needed?

Reply:
it could conflict with some macro

Reply:
https://github.com/microsoft/DiskANN/blob/main/CMakeLists.txt . here you also configure "NOMINMAX". but I could not use this way in our project, or else it will cause other compile errors

Reply:
@theantony Does this change affect you? Asking since you included the -DNOMINMAX in windows cmake options.

Reply:
Thanks @harsha-simhadri ! @yiylin What is the scenario where you're encountering the issue? Are you building DiskANN as part of your pipeline and the build breaks? Or are you using pre-built DiskANN as nuget package or similar and are getting build breaks when using DiskANN headers?

Reply:
I use this repo as submodule, then include this header to our project. Do you have the pre-built DiskANN nuget package?

Reply:
We build DiskANN as part of another nuget, it's not an "official" nuget.

I think these #undef calls are not the right place in DiskANN, they may cause issues with projects using DiskANN.

If you're building DiskANN as submodule, won't you get `NOMINMAX` defined so you won't have this issue? Or if you're using your custom build setting, can you add `NOMINMAX` just for DiskANN?

Then if you really need to #undef min or max, you can do it before including DiskANN headers and not impact other users of DiskANN.

Reply:
fine. if you have concerns, then I could remove this. I will try other ways to fix the issue in our side.

Reply:
Thanks! Looks good.